verification/spec/connection-handshake/chm.tla

```diff
@@ -0,0 +1,237 @@
+      Stores the current state of this connection. It has one of the
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+
+    - connection -- a connection with two ends
+      Stores the connection.
+
+    - chainHeight -- a height
+      Stores the height of the chain running alongside the CHM.
+ ***************************************************************************)
+ConnectionHandshakeModules ==
+    [
+        connectionState : ConnectionStates,
+        connection : Connections,
+        chainHeight : Heights
+        \* light client, consensus state, etc.. 
+    ]
+
+
+
+(******************************** Messages ********************************
+These messages are connection handshake specific.
+ ***************************************************************************)
+ConnectionHandshakeMessages ==
+    [type : {"ConnOpenInit"}, 
+     connection : Connections]
+\*    \union
+\*    [type : {"ConnOpenTry"},
+\*        connection : Connections,
+\*        stateProof : Proofs, 
+\*        consensusHeight : Proofs]
+
+
+
+(***************************************************************************
+ Connection handshake message handlers.
+***************************************************************************)
+
+\* The chModule handles a "ConnOpenInit" message.
+connectionOpenInit(chm, initMsg) ==
+  \* If we're in the right state.
+  IF chm.connectionState = "UNINIT" THEN
+         [initCHM |-> [chm EXCEPT !.connection = initMsg.connection,
+                                  !.connectionState = "INIT"],
+          msg |-> [type |-> "ConnOpenTry",
+                   connection |-> initMsg.connection]]
+                   \* TODO: Proofs go here.                   
+   \* Otherwise, do not update the chain
+   ELSE [initCHM |-> chm,
+         msg |-> noMsg ]
+
+handleMsgConnectionOpenInit ==
+ /\ inMsg.type = "ConnOpenInit"
+ /\ LET res == connectionOpenInit(chModule, inMsg) IN
+      /\ chModule' = res.initCHM
+      /\ outMsg' = res.msg
+      /\ inMsg' = noMsg \* The chm consumed its inbound message.
+
+(***************************************************************************
+ Connection Handshake Module actions.
+ ***************************************************************************)
+
+\* Advance the height of the chain if MaxHeight is not yet reached.
+advanceChainHeight ==
+    /\ chModule.chainHeight < MaxHeight
+    /\ chModule' = [chModule EXCEPT
+                        !.chainHeight = chModule.chainHeight + 1]
+    /\ UNCHANGED <<outMsg, inMsg>>
+
+
+(***
+  * Initial value for a connection end.
+  **)
+ConnectionEndInitValue ==
+    [connectionID |-> Null,
+     clientID |-> Null]
+
+
+(***
+  * Initial values for a connection.
+  **)
+ConnectionInitValue ==
+    [localEnd |-> ConnectionEndInitValue,
+     remoteEnd |-> ConnectionEndInitValue]
+
+CHModuleInitValue ==
+    [chainHeight |-> 1,
+     connectionState |-> "UNINT",
+     connection |-> ConnectionInitValue ]
```

Comment:
We should just define NilConnection as operator and that should be initial value of the connection field.